modules/expo-localization/src/Localization.js

```diff
@@ -0,0 +1,68 @@
+// @flow
+import { NativeModulesProxy, EventEmitter, Platform } from 'expo-core';
+const { ExpoLocalization } = NativeModulesProxy;
+
+type NativeEvent = {
+  locale: string,
+  locales: Array<string>,
+  timezone: string,
+  isoCurrencyCodes: ?Array<string>,
+  country: ?string,
+  isRTL: boolean,
+};
+
+type Listener = (event: NativeEvent) => void;
+
+type Subscription = {
+  remove: () => void,
+};
+
+class LocalizationModule {
+  _nativeEmitter: ?EventEmitter;
+  _nativeEventName: string = 'Expo.onLocaleUpdated';
+
+  locale: string;
+  locales: Array<string>;
+  timezone: string;
+  isoCurrencyCodes: ?Array<string>;
+  country: ?string;
+  isRTL: boolean;
+
+  constructor() {
+    this._syncLocals(ExpoLocalization);
+    if (Platform.OS === 'android') {
+      this._nativeEmitter = new EventEmitter(ExpoLocalization);
+      // TODO: Bacon: Need a way to destory this listener
```

Comment:
Is this a big problem? Have you experienced any memory leaks when refreshing an app repeatedly? If this causes problems maybe instead of making this module listener-based it would be better to just make async `getCountryAsync()` methods?

I appreciate the elegance of this way, just wouldn't like to ship something leaking. 😅

Reply:
This is a comment from an older approach. This listener is almost just copy paste from sensors https://github.com/expo/expo-sensors/blob/master/sensor/DeviceSensor.js

Its hard to say if the app crashes when the language changes, you need to go to settings. Then when you come back the app reloads, but so does every other app. I'd say at this point nothing crazy is going on.

Reply:
Hm, the difference between this approach and the DeviceSensor’s one is that here we export a one instance of LocalizationModule that has registered as a listener of the native module and in DS we export a class, instances of which are able to add listeners to the native module (so with DS developer can remove listeners to the native module, with LocationModule he cannot).

It looks to me a bit like the first pitfall Krzysztof has described in his [blog post](https://blog.swmansion.com/hunting-js-memory-leaks-in-react-native-apps-bd73807d0fde), adding listener and not removing it anywhere… but I may be wrong on that! Maybe when the app is destroyed its listeners are removed too, automatically. Just wanted to draw special attention to this part of code. 🙂

Reply:
I'm noticing that this has strange behavior on Expo because the app partially resets, and so does the experience. Just having the callback causes the app to crash if you change the language twice. 
I'm thinking we just remove all the callback functionality in favor of async stuff 🤷‍♀️